test/StringSerialization.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <microcdr/microcdr.h>
+
+#define BUFFER_LENGTH 1024
+#define ARRAY_LENGTH 255
+
```

Comment:
Note: make a test that check deserialize fails if user buffer is less that serialized string.

Reply:
Done!